writers_rollingfilewriter.go

```diff
@@ -138,7 +140,7 @@ var compressionTypes = map[rollingArchiveType]compressionType{
 			if err != nil {
 				return err
 			}
-			return createGzip(archiveName, tar)
+			return createGzip(archiveName, bytes.NewReader(tar))
```

Comment:
I played with the code a little more yesterday, and I think you're right. It's too hard to return the appropriate readers without buffering them to disk (see our tar.Header computation for example). I'm going to take a stab at replacing the `archive` function with the following interface:
```
type archiver interface {
	WriteHeader(fname string, fi *os.FileInfo) error
	Write(p []byte) (n int, err error)
}
```
The difficulty here is catering to multiple header formats (tar and zip). Ideally want to allow users to reuse the headers that they read in `unarchive`, but then we lose type safety. I'll keep thinking about this.